packages/passwordless-auth-construct/src/custom-auth/custom_auth_service.ts

```diff
@@ -213,11 +251,20 @@ export class CustomAuthService {
   private validateDeliveryCodeDetails = (
     event: CreateAuthChallengeTriggerEvent
   ): CodeDeliveryDetails => {
+    const previousChallenge = event.request.session.slice(-1)[0];
+    const previousDeliveryMedium: string | undefined =
+      previousChallenge.challengeMetadata?.includes('deliveryMedium')
+        ? JSON.parse(previousChallenge.challengeMetadata).deliveryMedium
+        : undefined;
```

Comment:
```suggestion
JSON.parse(previousChallenge.challengeMetadata).deliveryMedium
```
It will return `undefined` if the key is not found.